interfaces/cython/cantera/ck2yaml.py

```diff
@@ -1375,9 +1375,15 @@ def load_extra_file(self, path):
         """
         Load YAML-formatted entries from ``path`` on disk.
         """
-        yaml_ = yaml.YAML()
-        with open(path, 'rt', encoding="utf-8") as stream:
-            yml = yaml_.load(stream)
+        try:
+            yaml_ = yaml.YAML(typ="rt")
+            with open(path, 'rt', encoding="utf-8") as stream:
+                yml = yaml_.load(stream)
+        except yaml.constructor.ConstructorError:
+            with open(path, "rt", encoding="utf-8") as stream:
+                # Ensure that the loader remains backward-compatible with legacy
+                # ruamel.yaml versions (prior to 0.17.0).
+                yaml.round_trip_load(stream)
```

Comment:
I'd suggest passing a `pathlib.Path` instance directly to `load()`, rather than using the context manager. I'm not sure what version started supporting that, but hopefully something that's recent enough...

Also, in the last line of the `except` clause, you don't assign the return value to a variable.

Reply:
whoops - fixed the assignment. Regarding `pathlib.Path`, I don't think that the function receives the correct argument (from what I see it's a string). For the time being, I hope to just fix #1059 as I inadvertently broke the build for older `ruamel.yaml` versions - getting rid of `os` in `ck2yaml` is a slightly larger project (it currently doesn't use `pathlib` at all).